resilience4j-kotlin/src/main/kotlin/io/github/resilience4j/kotlin/timelimiter/TimeLimiter.kt

```diff
@@ -45,10 +47,12 @@ suspend fun <T> TimeLimiter.executeSuspendFunction(block: suspend () -> T): T =
         }
     } catch (t: Throwable) {
         if (isCancellation(coroutineContext, t)) {
-            onError(t)
-        } else {
-            onSuccess()
+            val timeoutException = TimeoutException(t.message)
```

Comment:
I think we should not loose the root cause here, or?

Can you use `TimeoutException(t)`.
Does that work in Kotlin?

Reply:
Do we need to add `@Throws(TimeoutException::class)` to the function?

Reply:
TimeoutException only has two constructors:
```
    public TimeoutException() 
    public TimeoutException(String message) 
```

Not really any way to add a cause or anything to it. That is a bit unfortunate though.  and I agree a stack trace would be very valuable.

Of course, we can always use extension since TimeoutException extends Exception , which does have a 'cause' variable.

Reply:
It's a long time since I used Kotlin: Why can't you use the constructors of the parent Java class in Kotlin?

Reply:
True.
We could do

```
val timeoutException = TimeoutException(t.message)
timeoutException.stackTrace = t.stackTrace
```